streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -230,6 +230,8 @@ public void onPartitionsRevoked(final Collection<TopicPartition> assignment) {
                 // suspend active tasks
                 suspendTasksAndState();
             } catch (final Throwable t) {
+                // TODO: when this happens, we should still make sure the standby / active tasks list gets updated before returning
```

Comment:
@guozhangwang is this not true already? In `suspendTasksAndState` we only throw an exception at the very end of the method.
Also, any tasks that failed to be suspended are closed with `task.close(false)`